src/Compilers/VisualBasic/Portable/VisualBasicParseOptions.vb

```diff
@@ -174,16 +187,16 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         ''' <param name="version">The parser language version.</param>
         ''' <returns>A new instance of VisualBasicParseOptions if different language version is different; otherwise current instance.</returns>
         Public Shadows Function WithLanguageVersion(version As LanguageVersion) As VisualBasicParseOptions
-            version = version.MapLatestToVersion()
-            If version = _languageVersion Then
+            If version = _specifiedLanguageVersion Then
                 Return Me
             End If
 
-            If Not version.IsValid Then
+            Dim effectiveVersion = version.MapSpecifiedToEffectiveVersion()
+            If Not effectiveVersion.IsValid Then
                 Throw New ArgumentOutOfRangeException(NameOf(version))
             End If
 
-            Return New VisualBasicParseOptions(Me) With {._languageVersion = version}
+            Return New VisualBasicParseOptions(Me) With {._specifiedLanguageVersion = version, ._languageVersion = effectiveVersion}
```

Comment:
That would be adding code as a premature optimization, a particularly bad practice that we tend to avoid [http://c2.com/cgi/wiki?PrematureOptimization](http://c2.com/cgi/wiki?PrematureOptimization)

---
In reply to: [74505258](https://github.com/dotnet/roslyn/pull/12969#discussion_r74505258) [](ancestors = 74505258)